src/widgets/panel/panel.tsx

```diff
@@ -59,40 +59,51 @@ export type PanelCSSClasses = {
   footer?: string | string[];
 };
 
-export type PanelTemplates = {
+export type UnknownWidgetFactory = WidgetFactory<any, any, any>;
```

Comment:
If we export this, we can move it to `src/types/widget.ts` and use it in both places.

Reply:
For now, `UnknownWidgetFactory` is used only inside this file. I think it should be alright to put it here. However, I've wondering when to export a type and when to not? Do we export all the types that are related to public APIs?

Reply:
What I mean is that we use this also here so we might as well reuse this type: https://github.com/algolia/instantsearch.js/pull/4527/files#diff-6b1f4a847fe51829e361b8d076cfff7b761fb616d83cabdfea076a08524f5c82R20.

Reply:
Alright. I forgot about the component. https://github.com/algolia/instantsearch.js/pull/4527/commits/44643f9383765feba496e3507cb5db033501f88f